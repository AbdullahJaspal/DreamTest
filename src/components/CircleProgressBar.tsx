import React from 'react';
import {View, StyleSheet, Text} from 'react-native';
import Svg, {Circle} from 'react-native-svg';

interface CircleProgressBarProps {
  progress: number;
  radius: number;
  strokeWidth: number;
}

const CircleProgressBar: React.FC<CircleProgressBarProps> = ({
  progress,
  radius,
  strokeWidth,
}) => {
  const circumference = 2 * Math.PI * radius;
  const progressOffset = circumference - (progress / 100) * circumference;

  return (
    <View style={styles.container}>
      <Svg width={radius * 2} height={radius * 2}>
        <Circle
          cx={radius}
          cy={radius}
          r={radius - strokeWidth / 2}
          stroke="lightgray"
          strokeWidth={strokeWidth}
          fill="transparent"
        />
        <Circle
          cx={radius}
          cy={radius}
          r={radius - strokeWidth / 2}
          stroke={'orange'}
          strokeWidth={strokeWidth}
          fill="transparent"
          strokeDasharray={`${circumference}, ${circumference}`}
          strokeDashoffset={progressOffset}
          strokeLinecap="round"
        />
      </Svg>
      <Text style={styles.text}>{progress}%</Text>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    justifyContent: 'center',
    alignItems: 'center',
    position: 'absolute',
    top: 10,
  },
  text: {
    position: 'absolute',
    fontSize: 13,
    fontWeight: 'bold',
    color: '#fff',
    textAlign: 'center',
  },
});

export default CircleProgressBar;
