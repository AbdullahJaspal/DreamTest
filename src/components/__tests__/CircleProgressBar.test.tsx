import 'react-native';
import React from 'react';
import CircleProgressBar from '../CircleProgressBar';

import {it} from '@jest/globals';

import renderer from 'react-test-renderer';

it('CircleProgressBar renders correctly', () => {
  // Render the CText component
  const tree = renderer
    .create(
      <CircleProgressBar
        progress={0}
        radius={0}
        strokeWidth={0}></CircleProgressBar>,
    )
    .toJSON();

  // Compare the rendered output to the saved snapshot
  expect(tree).toMatchSnapshot();
});
