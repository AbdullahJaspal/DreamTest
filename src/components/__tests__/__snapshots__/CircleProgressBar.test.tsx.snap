// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CircleProgressBar renders correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "justifyContent": "center",
      "position": "absolute",
      "top": 10,
    }
  }
>
  <RNSVGSvgView
    bbHeight={0}
    bbWidth={0}
    focusable={false}
    height={0}
    style={
      {
        "backgroundColor": "transparent",
        "borderWidth": 0,
      }
    }
    width={0}
  >
    <RNSVGGroup
      fill={
        {
          "payload": 4278190080,
          "type": 0,
        }
      }
    >
      <RNSVGCircle
        cx={0}
        cy={0}
        fill={
          {
            "payload": 0,
            "type": 0,
          }
        }
        propList={
          [
            "fill",
            "stroke",
            "strokeWidth",
          ]
        }
        r={0}
        stroke={
          {
            "payload": 4292072403,
            "type": 0,
          }
        }
        strokeWidth={0}
      />
      <RNSVGCircle
        cx={0}
        cy={0}
        fill={
          {
            "payload": 0,
            "type": 0,
          }
        }
        propList={
          [
            "fill",
            "stroke",
            "strokeWidth",
            "strokeDasharray",
            "strokeDashoffset",
            "strokeLinecap",
          ]
        }
        r={0}
        stroke={
          {
            "payload": 4294944000,
            "type": 0,
          }
        }
        strokeDasharray={
          [
            "0",
            "0",
          ]
        }
        strokeDashoffset={null}
        strokeLinecap={1}
        strokeWidth={0}
      />
    </RNSVGGroup>
  </RNSVGSvgView>
  <Text
    style={
      {
        "color": "#fff",
        "fontSize": 13,
        "fontWeight": "bold",
        "position": "absolute",
        "textAlign": "center",
      }
    }
  >
    0
    %
  </Text>
</View>
`;
